src/coord/src/coord.rs

```diff
@@ -316,9 +316,23 @@ where
             }
         }
 
-        for (id, name, _item) in catalog_entries {
+        for (id, name, item) in catalog_entries {
             // Mirror each recovered catalog entry.
             coord.report_catalog_update(id, name.to_string(), 1);
+
+            match item {
+                item @ CatalogItem::Source(_)
+                | item @ CatalogItem::Table(_)
+                | item @ CatalogItem::View(_) => {
+                    coord.update_mz_columns_catalog_view(
+                        item.desc().unwrap(),
```

Comment:
The set of things that have columns is precisely the set of things which have a descriptor, so I think you can simplify this a bit!

```rust
            if let Ok(desc) = item.desc() {
                coord.update_mz_columns_catalog_view(desc, &name.to_string(), entry.id(), 1);
            }
```